src/main/java/VASSAL/build/module/map/PieceMover.java

```diff
@@ -1112,10 +1109,19 @@ private void drawDragImage(BufferedImage image, Component target,
           else {
             stackCount = 0;
           }
-
+          
           final int x = EXTRA_BORDER - boundingBox.x + pos.x + offset.x;
           final int y = EXTRA_BORDER - boundingBox.y + pos.y - offset.y;
+          
+          String owner = "";
+          if (piece.getParent() instanceof Deck) {
+            owner = (String)piece.getProperty(Properties.OBSCURED_BY);
+            piece.setProperty(Properties.OBSCURED_BY, ((Deck) piece.getParent()).isFaceDown() ? Deck.NO_USER : null);
+          }
           piece.draw(g, x, y, map == null ? target : map.getView(), zoom);
+          if (piece.getParent() instanceof Deck) {
+            piece.setProperty(Properties.OBSCURED_BY, owner);
+          }
```

Comment:
Doing this in a draw method is truly horrible, but I don't see a good alternative right now without some _major_ refactoring.

Reply:
Yeah it's really less the Deck's implementation that is bad here than it is the Obscured implementation. We can look at a deeper fix if we want, but this should "hold the water back" for now.

Reply:
If you want to REALLY laugh and laugh, look at the draw method for **Deck** -- 

It does this:
      Object owner = top.getProperty(Properties.OBSCURED_BY);
      top.setProperty(Properties.OBSCURED_BY, faceDown ? NO_USER : null);

And this time it DOESN'T restore the state. So I will revise my previous statement to "It's really less Deck's appalling implementation that's affecting us here..." :O

Reply:
That probably causes subtle occasional funky behavior by cards in decks, by the way. The whole reason this ended up getting brought to Brent's attention at all was because I went looking for the source of some 3.2.17 funky deck behavior, and found Brent's patch had "partially" resolved it.

Reply:
Geez, maybe fix that, too?

Reply:
Okay I think a downstream fix might be along these lines:
#1 - Obscurable.java (only) gets a special draw method with an extra parameter to allow overriding the obscured state
#2 - Deck.java and PieceMover.java utilize that parameter instead of changing the game state.

Reply:
(It's funny how GH makes all your number signs into issue links. I was wonder what these had to do with the first two PRs.)

Reply:
Yeah I'll pull together a little mini-refactor patch and submit it for you to have a look at.

Reply:
(Now that this got less urgent and became more of an interesting/educational housecleaning project)

Reply:
PR #100 (TIL how to make PR autolink, haha)

Reply:
The Drag image is only created once per drag, so the overhead isn't significant. The one in Deck.draw() is worse. It does restore the state though at line 993. At a minimum, the Deck image should be cached, it does not change that often compared to the number of times the Deck is redrawn. It only changes when the top card changes.

I think Yan already has a Deck rewrite in his sights, but I think this will be difficult to achieve without introducing significant bugs and subtle behaviour changes.